src/statscreen.c

```diff
@@ -0,0 +1,3064 @@
+    sub_8004D5C(gBG0TilemapBuffer + TILEMAP_INDEX(1, 17), TEXT_COLOR_GOLD, 0x22, 0x23);
+    sub_8004B0C(gBG0TilemapBuffer + TILEMAP_INDEX(5, 17), TEXT_COLOR_GOLD, 0x16);
+
+    // Display level
+    DrawDecNumber(gBG0TilemapBuffer + TILEMAP_INDEX(4, 15), TEXT_COLOR_BLUE,
+        gStatScreen.unit->level);
+
+    // Display exp
+    DrawDecNumber(gBG0TilemapBuffer + TILEMAP_INDEX(7, 15), TEXT_COLOR_BLUE,
+        gStatScreen.unit->exp);
+
+    // Display current hp
+
+    if (GetUnitCurrentHp(gStatScreen.unit) > 99)
+    {
+        // Display '--' if current hp > 99
+        sub_8004D5C(gBG0TilemapBuffer + TILEMAP_INDEX(3, 17), TEXT_COLOR_BLUE,
+            0x14, 0x14);
+    }
+    else
+    {
+        // Display current hp
+        DrawDecNumber(gBG0TilemapBuffer + TILEMAP_INDEX(4, 17), TEXT_COLOR_BLUE,
+            GetUnitCurrentHp(gStatScreen.unit));
+    }
+
+    // Display max hp
+
+    if (GetUnitMaxHp(gStatScreen.unit) > 99)
+    {
+        // Display '--' if max hp > 99
+        sub_8004D5C(gBG0TilemapBuffer + TILEMAP_INDEX(6, 17), TEXT_COLOR_BLUE,
+            0x14, 0x14);
+    }
+    else
+    {
+        // Display max hp
+        DrawDecNumber(gBG0TilemapBuffer + TILEMAP_INDEX(7, 17), TEXT_COLOR_BLUE,
+            GetUnitMaxHp(gStatScreen.unit));
+    }
+}
+
+void sub_8086FAC(void)
+{
+    struct UnitUsageStats* stats = BWL_GetEntry(gStatScreen.unit->pCharacterData->number);
+
+    if (!stats)
+        return;
+
+    if (gUnknown_0202BCB0.gameStateBits & 0x40)
+        return;
+
+    if (gUnknown_0202BCF0.chapterStateBits & CHAPTER_FLAG_3)
+        return;
+
+    if (gUnknown_0202BCF0.chapterStateBits & CHAPTER_FLAG_7)
+        return;
+
+    if (IsFirstPlaythrough() == TRUE)
+        return;
+
+    if (UNIT_FACTION(gStatScreen.unit) != FACTION_BLUE)
+        return;
+
+    Text_Clear(&gStatScreen.text[STATSCREEN_TEXT_BWL]);
+
+    // Draw B label
+    Text_InsertString(&gStatScreen.text[STATSCREEN_TEXT_BWL],
+        0, TEXT_COLOR_GOLD, GetStringFromIndex(0x51F));
+
+    // Draw W label
+    Text_InsertString(&gStatScreen.text[STATSCREEN_TEXT_BWL],
+        32, TEXT_COLOR_GOLD, GetStringFromIndex(0x520));
+
+    // Draw L label
+    Text_InsertString(&gStatScreen.text[STATSCREEN_TEXT_BWL],
+        64, TEXT_COLOR_GOLD, GetStringFromIndex(0x521));
+
+    // Display labels
+    Text_Draw(&gStatScreen.text[STATSCREEN_TEXT_BWL],
+        gBmFrameTmap0 + TILEMAP_INDEX(3, 14));
+
+    // Display Battle Amt
+    sub_8004B88(gBmFrameTmap0 + TILEMAP_INDEX(3 + sub_80AEBEC(stats->battleAmt), 14),
+        TEXT_COLOR_BLUE, stats->battleAmt);
+
+    // Display Win Amt
+    sub_8004B88(gBmFrameTmap0 + TILEMAP_INDEX(7 + sub_80AEBEC(stats->winAmt), 14),
+        TEXT_COLOR_BLUE, stats->winAmt);
+
+    // Display Loss Amt
+    sub_8004B88(gBmFrameTmap0 + TILEMAP_INDEX(11 + sub_80AEBEC(stats->lossAmt), 14),
+        TEXT_COLOR_BLUE, stats->lossAmt);
+}
+
+#ifdef NONMATCHING
+
+void DrawStatScreenBar(int num, int x, int y, int base, int total, int max)
```

Comment:
Whats nonmatching about this?

Reply:
Looking at it closely again and I got it to match.
It was, of course, something dumb (the `total` argument needed to be loaded into a register which I initially thought was something to do with the diff variable being weird but actually that wasn't it).